### ISSUE TITLE
Build docker images on windows

### ISSUE THREAD

**Issue by avgustinmm** [eclipse/hono#3372]
When you build docker images with current Hono version on windows it builds corrupted unusable images - the start shell script doesn't have execute permissions and could not be run.
This could be fixed (at least for test purposes) by changing fabric8 plugin assembly/permissions from default keep to auto.
Then all files in the image would have exec permissions (which could be a security issue?) and images are usable - at least for testing purposes.

**Comment by avgustinmm** [comment]
I volunteer to fix that in the proposed way - adding support for docker images build with fabric8 plugin assembly/permissions `auto` mode

**Comment by calohmn** [comment]
Sounds good, if it can be done in a way that doesn't affect the standard linux build.

**Comment by calohmn** [comment]
Context: https://github.com/moby/moby/issues/11047

**Comment by calohmn** [comment]
@avgustinmm Instead of setting assembly permissions to `auto`, I guess we could alternatively just set the executable permission explicitly on the `/opt/hono/run-java.sh` file in the Dockerfile.
This can be done by adding 
```
<runCmds>
  <run>chmod 755 /opt/hono/run-java.sh</run>
</runCmds>
```
in the `docker-maven-plugin` configuration, alongside the `<entryPoint>` element.
Could you test this?

**Comment by avgustinmm** [comment]
@calohmn it seems to work with `runCmd`. Do you prefer this approach?
advantages would be 
- no need to configure anything on windows in order to run
- the rest (all but run script) would not have exec permissions as it is the case when build with `auto` on windows
- so the images built on windows might be production ready (however having that it seems there are unexpected differences with linux - who knows)
Disadvantages:
- this runCmd would be included when building on linux also. there seem to be an extra unnecessary step
I've no strong preference.

**Comment by calohmn** [comment]
I would prefer the `runCmd` approach. Much smaller change overall and works out of the box on windows.
I think the extra `RUN chmod 755 /opt/hono/run-java.` line in the Dockerfile doesn't hurt.

Btw, have you tested building the native images on windows?

**Comment by avgustinmm** [comment]
OK, I will change PR with the `runCmd` approach. After some more thinking about I also decided that it would be better approach. 
Haven't tested native images on windows

**Comment by calohmn** [comment]
Fixed via #3374.
